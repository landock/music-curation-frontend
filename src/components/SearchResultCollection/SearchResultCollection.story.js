import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { linkTo } from '@storybook/addon-links';
import SearchResultCollection from '.';
import { tracks } from '../../fixtures/test_data';

storiesOf('SearchResultCollection', module).add('default', () => (
  <SearchResultCollection searchResults={tracks} />
));
