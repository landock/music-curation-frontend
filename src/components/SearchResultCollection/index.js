import React, { Component } from 'react';
import PropTypes from 'prop-types';
import SearchResultCard from '../../containers/SearchResultCard';
import { Item, Header } from 'semantic-ui-react';
import { SortableContainer, arrayMove } from 'react-sortable-hoc';

const SortableList = SortableContainer(({ searchResults }) => {
  return (
    <Item.Group>
      {searchResults.map((result, index) => (
        <SearchResultCard key={result.id} index={index} {...result} />
      ))}
    </Item.Group>
  );
});

export default class SearchResultCollection extends Component {
  constructor(props) {
    super(props);
    this.state = {
      searchResults: props.searchResults,
    };
  }
  onSortEnd = ({ oldIndex, newIndex }) => {
    const { searchResults } = this.state;

    this.setState({
      searchResults: arrayMove(searchResults, oldIndex, newIndex),
    });
  };

  render() {
    const { searchResults } = this.state;

    return searchResults ? (
      <div>
        <Header>Search Results</Header>
        <p>{searchResults.length} Results</p>
        <SortableList
          searchResults={searchResults}
          onSortEnd={this.onSortEnd}
          useDragHandle={true}
        />
      </div>
    ) : (
      <p>no search results</p>
    );
  }
}
