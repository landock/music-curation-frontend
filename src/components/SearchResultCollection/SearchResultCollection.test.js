
import React from 'react'
import { shallow, mount, render } from 'enzyme';
import renderer from 'react-test-renderer'
import SearchResultCollection from '.'

it('SearchResultCollection: default', () => {
  const component = renderer.create(<SearchResultCollection />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})