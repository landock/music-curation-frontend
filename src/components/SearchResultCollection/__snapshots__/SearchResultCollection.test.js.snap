// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResultCollection: default 1`] = `
<p>
  no search results
</p>
`;
